### PR TITLE
Add sfdp pattern for huggingface `BertForQuestionAnswering` inference model 

### LINKED ISSUE #126057: [Triton] Add sdpa pattern support for `BertForQuestionAnswering` model
### 🚀 The feature, motivation and pitch

In a `BertForQuestionAnswering` model, the core calculation process of the `BertSelfAttention.forward` function can be simplified as:
```python
key_layer = self.transpose_for_scores(self.key(hidden_states))
value_layer = self.transpose_for_scores(self.value(hidden_states))
query_layer = self.transpose_for_scores(mixed_query_layer)
attention_scores = torch.matmul(query_layer, key_layer.transpose(-1, -2))
attention_scores = attention_scores / math.sqrt(self.attention_head_size)
if attention_mask is not None:
    attention_scores = attention_scores + attention_mask
attention_probs = nn.functional.softmax(attention_scores, dim=-1)
attention_probs = self.dropout(attention_probs)
context_layer = torch.matmul(attention_probs, value_layer)
```
During inference, `dropout` layers will be treated as a clone layers. But the graph cannot match the existing pattern.
I hope `torch.compile` can support this static graph.

### Alternatives

_No response_

### Additional context

_No response_

cc @ezyang @msaroufim @bdhirsh @anijain2305 @chauhang @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire

### PR DESCRIPTION
**Pull request by hoshibara**
Fixes #126057

Patterns such as `_sfdp_pattern_14` do not support this inference graphs.
This PR adds graph support for `BertForQuestionAnswering` inference model.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/fuse_attention.py [hunk 1]
```diff
@@ -536,6 +536,32 @@ def _sfdp_replacement_19(query, key, value, causal_mask, attn_mask, dropout_p):
     )
 
 
+def _sfdp_pattern_20(query, key, value, attn_mask):
+    # For huggingface bert
```
Comment (jgong5) [inline torch/_inductor/fx_passes/fuse_attention.py:540]: please check if we can merge the logic with pattern_16 to avoid dedup.

Comment (hoshibara) [reply torch/_inductor/fx_passes/fuse_attention.py:540]: I tried merging pattern16 but I failed.
Pattern 16 cannot match this model.
